internal/healthcheck/checkers.go

```diff
@@ -0,0 +1,98 @@
+package healthcheck
+
+import (
+	"context"
+	"sync"
+	"time"
+
+	"github.com/canonical/identity-platform-login-ui/internal/logging"
+	"go.opentelemetry.io/otel/trace"
+)
+
+type CheckFunction func(context.Context) (bool, error)
+
+type CheckerInterface interface {
+	Start()
+	Stop()
+	Status() bool
+}
+
+type Checker struct {
+	f      CheckFunction
+	ticker *time.Ticker
+	status bool
+
+	// goroutine control
+	wg sync.WaitGroup
+
+	// mutex
+	mutex sync.RWMutex
+
+	shutdownCh chan bool
+
+	tracer trace.Tracer
+	logger logging.LoggerInterface
+}
+
+func (c *Checker) Start() {
+	c.wg.Add(1)
+	go c.loop()
+}
+
+func (c *Checker) Stop() {
+	// send shutdown
+	c.shutdownCh <- true
+
+	c.wg.Wait()
+}
+
+func (c *Checker) Status() bool {
+	c.mutex.RLock()
+	defer c.mutex.RUnlock()
+
+	return c.status
```

Comment:
Why do we need the lock here? Can't we just use atomic store and load? (https://pkg.go.dev/sync/atomic#Bool)

Reply:
swapped to use the atomic.Bool